framework/behaviors/AttributeBehavior.php

```diff
@@ -118,6 +123,10 @@ public function evaluateAttributes($event)
             foreach ($attributes as $attribute) {
                 // ignore attribute names which are not string (e.g. when set by TimestampBehavior::updatedAtAttribute)
                 if (is_string($attribute)) {
+                    // skip attribute update when it is not empty and the corresponding mode is enabled
+                    if ($this->skipUpdateOnFilled && !empty($this->owner->$attribute)) {
```

Comment:
> can you use a native function instead of empty?
> Refactored that sub-condition to the $this->owner->$attribute !== ''.

the logic behind this should be made less complex. The current format is not self explaining enough and takes time to interpretate and has room for improvement.`skipUpdateOnFilled` in combination with a false comparison is less readable than a positive one with a variable name like `forceUpdateOnFilled`. Additionally, `onFilled` is  a bit vague. What is filled in the context? When is it about a change or about being empty? It would expect something more related to your original issue description.

> a check on $event->name is missing, like with $skipUpdateOnClean. This makes it confusing an less readable and error prone.
see https://github.com/Kolyunya/yii2/blob/8cee4ec59b9553b47adad4cee7e3828ee08123cc/framework/behaviors/AttributeBehavior.php#L113. The current code looks to be run in the 'insert' context also which could be confusing.